forest_lib/forest/aws.py

```diff
@@ -0,0 +1,194 @@
+existence, and if necessary trigger a download on demand.
+
+>>> file_loader.load_file('file.nc')
+
+Mount file system
+-----------------
+
+If a file system has been mounted using a POSIX-ish wrapper
+like **Goofys** then a file system interface can be used
+
+>>> file_loader = forest.aws.S3Mount('/mount/dir')
+>>> file_loader.load_file('file.nc')
+
+A mounted file system makes it quicker and easier for Forest
+to find and load data on demand as it only accesses the
+individual values needed to plot the data
+
+Application programming interface (API)
+---------------------------------------
+
+Forest can access data using Amazon credentials or
+download data from AWS using urllib
+
+"""
+import os
+import urllib.request
+
+
+# expose print to be patched
+print = print
+
+
+class S3Bucket(object):
+    """S3 bucket urllib interface
+
+    Loads files from **AWS** S3 bucket using :py:mod:`urllib`
+    module from the Python standard library, specifically
+    ``urllib.requests``
+
+    :param server_address: Amazon server URL
+    :param bucket_name: name of bucket
+    :param download_directory: directory on local file system to
+                               store file(s)
+    """
+    def __init__(self,
+                 server_address,
+                 bucket_name,
+                 download_directory):
+        self.server_address = server_address
+        self.bucket_name = bucket_name
+        self.download_directory = download_directory
+
+    def file_exists(self, file_name):
+        """AWS file exists or downloaded file exists
+
+        .. note:: Only checks remote file existence
+                  to support peculiarities of :mod:`forest.data`
+
+        :param file_name: base name of file to be queried
+        :returns: logical indicating file existence
+        """
+        return self.remote_file_exists(file_name)
+
+    def remote_file_exists(self, file_name):
+        """Check file present in S3 bucket
+
+        :param file_name: base name of file to be queried
+        :returns: logical indicating file existence
+        """
+        return self.s3_file_exists(self.s3_url(file_name))
+
+    @staticmethod
+    def s3_file_exists(url):
+        """Check whether a remote file exists; return Bool.
+
+        Check whether a file at the remote location specified by remore
+        path exists by trying to open a url request.
+
+        :param remote_path: Path to check for file at.
+        :returns: logical indicating file existence
+        """
+        try:
+            _ = urllib.request.urlopen(remote_path)
+            print('file {0} found at remote location.'.format(remote_path))
+            return True
+        except urllib.error.HTTPError:
+            warning_msg1 = 'warning: file {0} NOT found at remote location.'
+            warning_msg1 = warning_msg1.format(remote_path)
+            print(warning_msg1)
+        return False
+
+    def local_file_exists(self, file_name):
```

Comment:
maybe make methods like this `_private` methods. Keep the interface the same as S3Mount and other methods `_private` as I feel that they are implementation details.